### DISCUSSION TITLE
Exclude files in global search

### DISCUSSION THREAD

**Discussion by FerroMeow** [helix-editor/helix#7631] [General] [answered]
### The problem
Working with global search would be more useful if we could exclude files to be searched.

The main motivation for this is performance - The editor freezes for minutes trying to search every single file in the system. This task is possible in other code editors, like VS Code.

Another benefit would be less clutter while searching. For example, if we're searching for all matches of word `class` in the code, we probably don't need the matches from `*.md` files

### Ideas regarding a possible solution
1. The ability to whitelist files for global search - if not matched, don't search
2. The ability to blacklist files for global search search - if matched, don't search
3. defining these by glob or regex syntax - regex would be a more robust solution I think

**Comment by pascalkuthe** [comment, marked as answer]
filtering paths using fuzzy matching in global search in case you want to only see certain files is already possible that won't affect performance tough. 

We want to make the selector streaming the future tough so that would improve performance. Helix however already respects ignore/git ignore files so a large number of build files can already be ignored that way. I think the use cases that you genuinely have millions of files that you don't want to ignore and can't wait to stream in is small and then you should probably use cli tools like rigprep and fd that are purpose-built.

**Comment by deadchatxd** [comment]
not directly related to the problem, but to silence some results we can search like this  `class %p !.md`, using %p to filter path results, as stated in https://helix-editor-tutorials.com/tutorials/using-helix-global-search/

that's just for people like me, who are new to helix and trying to get things working
